interfaces/builtin/serial_port.go

```diff
@@ -109,23 +164,63 @@ func (iface *SerialPortInterface) PermanentPlugSnippet(plug *interfaces.Plug, se
 
 // ConnectedPlugSnippet returns security snippet specific to the plug
 func (iface *SerialPortInterface) ConnectedPlugSnippet(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+
+	slotPath, slotOk := slot.Attrs["path"].(string)
+	if !slotOk || slotPath == "" {
+		// don't have path attribute so must be using Udev tagging
+
+		switch securitySystem {
+		case interfaces.SecurityAppArmor:
+			// Wildcarded apparmor snippet as the cgroup will restrict down to the
+			// specific device
+			return []byte("/dev/tty* rw,\n"), nil
+		case interfaces.SecurityUDev:
+			idVendor, vOk := plug.Attrs["vendor-id"].(string)
+			if !vOk {
+				return nil, fmt.Errorf("serial-port plug failed to find vendor-id attribute")
+			}
+			idProduct, pOk := plug.Attrs["product-id"].(string)
+			if !pOk {
+				return nil, fmt.Errorf("serial-port plug failed to find product-id attribute")
+			}
+			var udevSnippet bytes.Buffer
+			udevSnippet.WriteString(udevHeader + "\n")
+			for appName := range plug.Apps {
+				udevSnippet.WriteString(fmt.Sprintf(udevEntryPattern, idVendor, idProduct))
+				tag := fmt.Sprintf("snap_%s_%s", plug.Snap.Name(), appName)
+				udevSnippet.WriteString(fmt.Sprintf(udevEntryTagPattern, tag) + "\n")
+			}
```

Comment:
We shouldn't. We can add support for this easily later without breaking anything but for now I don't see any need to assign access to all devices from the same vendor or product type.